common/src/legacy_memory_region.rs

```diff
@@ -121,9 +138,11 @@ where
             let next_free = self.next_frame.start_address();
             let kind = match descriptor.kind() {
                 MemoryRegionKind::Usable => {
-                    if end <= next_free {
+                    if end <= next_free && end > self.start_frame.start_address() {
                         MemoryRegionKind::Bootloader
-                    } else if descriptor.start() >= next_free {
+                    } else if descriptor.start() >= next_free
+                        || end <= self.start_frame.start_address()
+                    {
```

Comment:
I would prefer to split this condition so that we can add a comment:

```suggestion
                    } else if descriptor.start() >= next_free {
                        MemoryRegionKind::Usable
                    } else if end <= self.start_frame.start_address() {
                        // treat regions that end before the minimum frame as usable too
```